src/ray/core_worker/core_worker.cc

```diff
@@ -1204,15 +1207,18 @@ Status CoreWorker::SubmitActorTask(const ActorID &actor_id, const RayFunction &f
   return status;
 }
 
-Status CoreWorker::KillTask(const ObjectID &object_id, bool force_kill) {
-  if (!object_id.CreatedByTask()) {
-    return Status::Invalid("Not created by a task.");
+Status CoreWorker::CancelTask(const ObjectID &object_id, bool force_kill) {
+  ActorHandle *h = nullptr;
+  if (!object_id.CreatedByTask() ||
+      GetActorHandle(object_id.TaskId().ActorId(), &h).ok()) {
```

Comment:
I think you can just check if `object_id.TaskId().ActorId().IsNil()` instead of getting the handle.

Reply:
The ActorId() is not nil for normal tasks :/